tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/files/S3StorageSpec.scala

```diff
@@ -206,6 +208,36 @@ class S3StorageSpec extends StorageSpec {
     }
   }
 
+  "Filenames with url-encodable characters" should {
+    "have an appropriate filename in S3" in {
+      val id   = genId()
+      val name = "name with spaces.txt"
+
+      for {
+        _ <- deltaClient.uploadFile[Json](
+               s"/files/$projectRef/$id?storage=nxv:$storageId",
+               "file contents",
+               ContentTypes.`text/plain(UTF-8)`,
+               name,
+               Coyote
+             )((json, response) => {
+               response.status shouldEqual StatusCodes.Created
+               json should have(filenameField(name))
```

Comment:
Can you also check if the location field is ok ?
The `Uri` from akka could bring surprises

Reply:
I eyeballed it and it's fine. Do you want that in the test also?

Reply:
Users use more location that the filename so if you can have an extra assertion to make sure that location == key, it would be great